app/testOnly/models/formsmodel/StartJourneyForm.scala

```diff
@@ -117,14 +117,15 @@ object StartJourneyForm {
               case TaxRegime.Epaye => payeTaxReferenceKey -> RandomDataGenerator.nextEpayeRefs()(Random)._3
               case TaxRegime.Vat   => vatTaxReferenceKey -> RandomDataGenerator.nextVrn()(Random)
             }
-            val dataWithDefaultValue: Map[String, String] =
-              if (data.get(taxReferenceKey).exists(_.nonEmpty)) data else data.updated(taxReferenceKey, defaultTaxRef.value)
-
-            val taxRef: TaxId = taxRegime match {
-              case TaxRegime.Epaye => EmpRef(dataWithDefaultValue.getOrElse(taxReferenceKey, defaultTaxRef.value))
-              case TaxRegime.Vat   => Vrn(dataWithDefaultValue.getOrElse(taxReferenceKey, defaultTaxRef.value))
+            val dataWithDefaultValue: String = {
```

Comment:
couldn't you do something like 
```scala
data.get(taxReferenceKey)
  .filter(_.nonEmpty)
  .map(s => taxRegimeMatch {
    case TaxRegime.Epaye => EmpRef(s)
    case TaxRegime.Vat => Vrn(s)
  })
  .getOrElse(defaultTaxRef)
```
then you don't need to update a map with a default value only then to do a `getOrElse` on it anyway

Reply:
will do